nav2_rviz_plugins/src/nav2_panel.cpp

```diff
@@ -161,6 +161,9 @@ Nav2Panel::Nav2Panel(QWidget * parent)
   state_machine_.setInitialState(pre_initial_);
   state_machine_.start();
 
+  // delay starting initial thread until state machine has started or a race occurs
+  QObject::connect(&state_machine_, SIGNAL(started()), this, SLOT(startThread()));
```

Comment:
This probably wouldn't happen (maybe only in debug), but should this line come before the start of the state machine in order to avoid another potential race condition (i.e. signal is emitted before connection is made)?

Reply:
Makes sense, done in my new commit